make/tools.mk

```diff
@@ -165,17 +164,19 @@ else
 endif
 
 # Define flags for (cross) build
-CFLAGS             += \
+NOARCH_CFLAGS      := \
   $(CFLAGS_EXT) \
   -fdata-sections \
   -ffunction-sections \
   -fno-asynchronous-unwind-tables \
   -pipe \
   -Wall
+CFLAGS             += $(NOARCH_CFLAGS) $(ARCHITECTURE_CFLAGS)
```

Comment:
I think it could be better to swap the order of substituted variables here so architecture-specific flags were the first ones.